src/app/editor/contextualListener.js

```diff
@@ -0,0 +1,93 @@
+'use strict'
+var SourceMappingDecoder = require('ethereum-remix').util.SourceMappingDecoder
+var AstWalker = require('ethereum-remix').util.AstWalker
+
+class ContextualListener {
+  constructor (api, events) {
+    this._api = api
+    this._index = {
+      ReferencedDeclarations: {},
+      Expressions: []
+    }
+    this._events = []
+
+    events.compiler.register('compilationFinished', (success, data, source) => {
+      this._stopWarning()
+      this._index = {
+        ReferencedDeclarations: {},
+        Expressions: []
+      }
+      if (success) {
+        this._buildIndex(data, source)
+      }
+    })
+    this.sourceMappingDecoder = new SourceMappingDecoder()
+    this.astWalker = new AstWalker()
+    setInterval(() => {
```

Comment:
Is this cancelled if we recompile in the meantime?

Reply:
(or change the text)?

Reply:
no but this is not really linked to the compilation event, the function in setInterval will be executed every 1s and check: if the position of the cursor changed and if there's a compilation result to work with.

Reply:
ok!